pkg/resources/machinecontroller/webhook.go

```diff
@@ -53,6 +54,24 @@ var (
 func WebhookDeploymentCreator(data machinecontrollerData) reconciling.NamedDeploymentCreatorGetter {
 	return func() (string, reconciling.DeploymentCreator) {
 		return resources.MachineControllerWebhookDeploymentName, func(dep *appsv1.Deployment) (*appsv1.Deployment, error) {
+			args := []string{
+				"-kubeconfig", "/etc/kubernetes/kubeconfig/kubeconfig",
+				"-logtostderr",
+				"-v", "4",
+				"-listen-address", "0.0.0.0:9876",
+				"-ca-bundle", "/etc/kubernetes/pki/ca-bundle/ca-bundle.pem",
+			}
+
+			externalCloudProvider := data.Cluster().Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider]
+			if externalCloudProvider {
+				args = append(args, "-node-external-cloud-provider=true")
+			}
+
+			featureGates := data.GetCSIMigrationFeatureGates()
+			if len(featureGates) > 0 {
+				args = append(args, "-node-kubelet-feature-gates", strings.Join(featureGates, ","))
```

Comment:
If you think this line is missing `"RotateKubeletServerCertificate=true"` then it's not. `"RotateKubeletServerCertificate=true"` defaulting to true is now part of the machine-controller (even when `-node-kubelet-feature-gates` is defined but without `RotateKubeletServerCertificate`.